package.json

```diff
@@ -15,32 +15,42 @@
   ],
   "devDependencies": {
     "benchmark": "^2.1.4",
-    "eslint": "3",
-    "eslint-config-egg": "3",
-    "istanbul": "0",
-    "koa": "2",
-    "mm": "^2.1.0",
-    "mocha": "^5.2.0",
+    "c8": "^7.0.0",
+    "coveralls": "^3.0.0",
+    "eslint": "^8.0.0",
+    "koa": "^2.0.0",
+    "mm": "^3.0.0",
+    "mocha": "^9.0.0",
     "mz-modules": "^2.0.0",
     "pedding": "^1.1.0",
     "uid-safe": "^2.1.3",
-    "should": "8",
-    "supertest": "^3.3.0"
+    "should": "^13.0.0",
+    "supertest": "^6.0.0"
   },
   "license": "MIT",
   "dependencies": {
     "crc": "^3.4.4",
-    "debug": "^3.1.0",
+    "debug": "^4.0.0",
     "is-type-of": "^1.0.0",
-    "uuid": "^3.3.2"
+    "uuid": "^8.0.0"
   },
   "engines": {
-    "node": ">=7.6"
+    "node": ">=10.0.0"
```

Comment:
This makes this a breaking change, which'd mean a new major version - could be worth trying to split this PR up into a couple of smaller ones so non-breaking changes can be landed?

Reply:
From memory, I think it was dev-deps that prompted the engines.node bump, though I didn't make particular note of it.

I've never been quite sure about whether the engines.node should relate to app + dependencies, or app + dependencies + devDependencies.

I think the only substantive changes (ignoring dev-deps) are uuid & debug; debug@4.3.2 specifies engines.node as >=6.0, uuid doesn't mention engines.node.

I think I bumped engines.node to 10+ since eslint required that, but whether engines.node should depend on a devDependency, I'm not sure!

To further complicate the issue, koa@2.13.1 specifies engines.node as >=12: but having just run a test, the koa-session `example.js` runs perfectly well on node@7.6.0.

If the koa-session maintainers feel it worthwhile, the deps could be bumped separately from the dev-deps, but perhaps engines.node should not depend on devDependencies, in which I was mistaken in bumping it up to 10.0.0 and it can validly stay on 7.6.0?

Reply:
I think it's ok to upgrade node to 10.0.0 since node<10 is already out of maintenance for a long time.